File: src/models/output/DAC_costing.jl
function gen_DAC_cost_df(m)

    vars = [
        "x_sorbent_total", "x_cost_sorbent",
        "x_air_adsorb_max",
        "x_cost_adsorb", "x_cost_DAC_TPC"
    ]

    df_DAC_costing = DataFrame(map(generate_row_map, vars))

    l0 = []
    for v in vars
        l0 = vcat(l0, value(m[Symbol(v)]))
    end
    push!(df_DAC_costing, l0)

    return df_DAC_costing

end

File: src/pyomo_model/output/DAC_costing.py
"""
Output module
Pengfei Cheng

Output DAC operation-related results.
"""

import pandas as pd
from pyomo.environ import value

def gen_DAC_cost_df(m):

    vars = [
        "x_sorbent_total",
        "x_air_adsorb_max",
    ]

    dict_list = []
    for var_name in vars:
        var = getattr(m, var_name)
        _tmp_dict = {k: value(var[k]) for k in var}
        dict_list.append(_tmp_dict)

    df = pd.DataFrame(dict_list).T
    df.columns = vars

    return df